openfl/component/aggregator/aggregator.py

```diff
@@ -545,13 +545,28 @@ def _straggler_cutoff_time_elapsed(self) -> None:
             # Check if minimum collaborators reported results
             self._end_of_round_with_stragglers_check()
 
+    def _check_tags(self, tags: tuple[str, ...], allowed_col: str) -> bool:
+        """
+        Check if all tags are either the allowed column or unauthorized.
+
+        This function verifies that no tag (except the explicitly allowed one)
+        belongs to the list of authorized columns.
+
+        Args:
+            tags (tuple[str, ...]): The set of tags to check.
+            allowed_col (str): The only authorized column allowed in the tags.
+
+        Returns:
+            bool: True if all tags are valid, False if an unauthorized column
+            (other than allowed_col) is found.
+        """
+        for tag in tags:
+            if tag in self.authorized_cols and tag != allowed_col:
```

Comment:
`authorized_cols` contains the names of collaborators authorized for the experiment (effectively, the entries in `cols.yaml`)

Reply:
Correct